static/scripts/rewards/gift-cards/list-gift-cards.ts

```diff
@@ -25,7 +26,10 @@ export async function initClaimGiftCard(app: AppState) {
   activateInfoSection.innerHTML = "";
 
   const retrieveOrderUrl = `${getApiBaseUrl()}/get-order?orderId=${getGiftCardOrderId(app.reward.beneficiary, app.reward.signature)}`;
-  const listGiftCardsUrl = `${getApiBaseUrl()}/list-gift-cards`;
+
+  const localTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+  const countries = ct.getCountriesForTimezone(localTimezone);
+  const listGiftCardsUrl = `${getApiBaseUrl()}/list-gift-cards?country=${countries[0].id}`;
```

Comment:
As far as I understand there might be multiple countries in a single timezone so this approach is not 100% accurate. Perhaps in the next iteration we can get country by user's ip address.

Reply:
This mostly worked well, I have tried for some countries where I thought timezone is same. But the code returns IANA timezone databases entry that appear to be unique for each country timezone. 

However, this is problematic when user has set some customized timezone. For example, I also have option to set timezone as UTC in my linux machine and it doesn't detect country then. 

Detecting via user's IP has also its drawback. VPNs, and adblockers blocking IP location service. Plus, we also have several requests going on the page load, this will an addition to that and it will be blocking request, because we need country before loading card list. 

How about we keep some dropdown in the UI to select your country? But still use timezone to auto detect.

Reply:
> As far as I understand there might be multiple countries in a single timezone so this approach is not 100% accurate. Perhaps in the next iteration we can get country by user's ip address.

I presume that the country name is being parsed from the timezone for example 

```
Here are some examples of IANA timezones:

1. **America/New_York**
2. **Europe/London**
3. **Asia/Tokyo**
4. **Australia/Sydney**
5. **Africa/Johannesburg**
6. **America/Los_Angeles**
7. **Europe/Berlin**
8. **Asia/Dubai**
9. **Pacific/Auckland**
10. **America/Sao_Paulo**
```

Reply:
Current approach is pretty fine:
1. Try to get user's country by ip
2. Otherwise get it from timezone

Reply:
> What about selecting the country based on the city name? Is that what is happening here?

Yes, that is how location detection via timezone is working.

Reply:
> A combined approach seems solid but I'm surprised there aren't open source tools/frameworks that already solved this problem fairly reliably that we can use?

There is one, it is [geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API). But it asks user for location and user can decline.  

It was easier to do it via timezone or IP. Geolocation API will need a different flow. We show a dummy card first, and ask user for location to get them appropriate card. If they decline, no card for them.